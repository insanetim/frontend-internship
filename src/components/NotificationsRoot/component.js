import Notification from './Notification'
import useContainer from './hook'

export default function NotificationsRoot() {
  const { notifications, hideNotification } = useContainer()

  return (
    <>
      {notifications.map(notification => (
        <Notification
          key={notification.id}
          {...notification}
          hideNotification={hideNotification}
        />
      ))}
    </>
  )
}
