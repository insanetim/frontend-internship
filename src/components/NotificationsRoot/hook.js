import { useDispatch, useSelector } from 'react-redux'

import { hideNotification as hideNotificationAction } from 'src/state/app/actions'
import { notificationsSelector } from 'src/state/app/selectors'

const useContainer = () => {
  const dispatch = useDispatch()
  const notifications = useSelector(notificationsSelector)

  const hideNotification = id => {
    dispatch(hideNotificationAction(id))
  }

  return { notifications, hideNotification }
}

export default useContainer
