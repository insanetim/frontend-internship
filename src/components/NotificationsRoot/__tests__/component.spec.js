import { shallow } from 'enzyme'

import NotificationsRoot from '../component'

const mockedHookData = {
  notifications: [
    {
      id: 'test/id',
      messageType: 'success',
      messageText: 'test/message',
      duration: 2.5
    }
  ],
  hideNotification: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

describe('NotificationsRoot component tests', () => {
  const component = shallow(<NotificationsRoot />)

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
