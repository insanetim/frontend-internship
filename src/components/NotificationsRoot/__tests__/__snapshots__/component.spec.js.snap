// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsRoot component tests matches snapshot 1`] = `
<Fragment>
  <Notification
    duration={2.5}
    hideNotification={[MockFunction]}
    id="test/id"
    key="test/id"
    messageText="test/message"
    messageType="success"
  />
</Fragment>
`;
